Index: src/autoapp/Service/SensorService.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/autoapp/Service/SensorService.cpp b/src/autoapp/Service/SensorService.cpp
--- a/src/autoapp/Service/SensorService.cpp	(revision f1976fe76cd576c01631fefde9baffceffb37567)
+++ b/src/autoapp/Service/SensorService.cpp	(revision cdf199263701207162c8a73c4fb16ed73cca72f7)
@@ -188,7 +188,7 @@
     auto * locInd = indication.add_gps_location();
 
     // epoch seconds
-    locInd->set_timestamp(this->gpsData_.fix.time.tv_sec);
+    locInd->set_timestamp(this->gpsData_.fix.time * 1e3);
     // degrees
     locInd->set_latitude(this->gpsData_.fix.latitude * 1e7);
     locInd->set_longitude(this->gpsData_.fix.longitude * 1e7);
@@ -229,7 +229,8 @@
 
             if ((this->gpsEnabled_) &&
                (gps_waiting(&this->gpsData_, 0)) &&
-               (gps_read(&this->gpsData_, nullptr, 0) > 0) &&
+               (gps_read(&this->gpsData_) > 0) &&
+               (this->gpsData_.status != STATUS_NO_FIX) &&
                (this->gpsData_.fix.mode == MODE_2D || this->gpsData_.fix.mode == MODE_3D) &&
                (this->gpsData_.set & TIME_SET) &&
                (this->gpsData_.set & LATLON_SET))
